app/post_metadata_test.go

```diff
@@ -189,7 +190,13 @@ func TestPreparePostForClient(t *testing.T) {
 
 		fileInfo.PostId = post.Id
 
-		clientPost := th.App.PreparePostForClient(post, false, false)
+		var clientPost *model.Post
+		assert.Eventually(t, func() bool {
+			p, nErr := th.App.Srv().Store.Post().Get(context.Background(), post.Id, false, false, false, "")
+			require.NoError(t, nErr)
+			clientPost = th.App.PreparePostForClient(p.Posts[post.Id], false, false)
+			return assert.ObjectsAreEqual([]*model.FileInfo{fileInfo}, clientPost.Metadata.Files)
+		}, time.Second, 100*time.Millisecond)
 
 		assert.Equal(t, []*model.FileInfo{fileInfo}, clientPost.Metadata.Files, "should've populated Files")
```

Comment:
Do we need to keep this? I thought the above assertion was sufficient.

Reply:
Yes, the idea is to provide a better output in that case, the previous error wasn't providing a nice output.